extensions/oauth/src/main/java/org/mule/extension/oauth2/internal/DeferredExpressionResolver.java

```diff
@@ -45,4 +46,15 @@ public DeferredExpressionResolver(MuleContext muleContext) {
     }
   }
 
+  public <T> String getExpression(ParameterResolver<T> expr) {
```

Comment:
ParameterResolver has a getExpression() method that returns the expression. What what would you do with it? In order to access the Event and the expression manager this should be a privileged extension (which is not ideal). If what yo need is deferred evaluation of that expression I don't understand why ParameterResolver doesn't do the trick